File: backend-spring/app-domain/src/main/java/apptemplate/domain/entities/Notification.java
package apptemplate.domain.entities;

import apptemplate.domain.enums.NotificationType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * Notification entity for real-time notifications.
 */
@Getter
@Setter
@NoArgsConstructor
public class Notification {

    private Long id;
    private Long userId;
    private String title;
    private String message;
    private NotificationType type;
    private String referenceId;
    private String referenceType;
    private boolean read;
    private LocalDateTime createdAt;

    /**
     * Creates a new notification.
     */
    public Notification(Long userId, String title, String message, NotificationType type,
                        String referenceId, String referenceType) {
        this.userId = userId;
        this.title = title;
        this.message = message;
        this.type = type;
        this.referenceId = referenceId;
        this.referenceType = referenceType;
        this.read = false;
        this.createdAt = LocalDateTime.now();
    }

    /**
     * Marks notification as read.
     */
    public void markAsRead() {
        this.read = true;
    }
}

File: backend/dotnet/clean-architecture/full/src/Core/App.Template.Domain/Entities/Notification.cs
using AppTemplate.Domain.Enums;

namespace AppTemplate.Domain.Entities;

public class Notification
{
    public long Id { get; private set; }
    public string UserId { get; private set; } = null!;
    public string Title { get; private set; } = null!;
    public string Message { get; private set; } = null!;
    public NotificationType Type { get; private set; }
    public string? ReferenceId { get; private set; } // Can be Process ID or Task ID
    public string? ReferenceType { get; private set; } // "Process" or "Task"
    public bool IsRead { get; private set; }
    public DateTime CreatedAt { get; private set; }

    private Notification() { }

    public Notification(string userId, string title, string message, NotificationType type, string? referenceId = null, string? referenceType = null)
    {
        UserId = userId;
        Title = title;
        Message = message;
        Type = type;
        ReferenceId = referenceId;
        ReferenceType = referenceType;
        IsRead = false;
        CreatedAt = DateTime.UtcNow;
    }

    public void MarkAsRead()
    {
        IsRead = true;
    }
}
